File: src/main/java/org/dedda/bratwurst/lang/ObjectCreation.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
public class ObjectCreation extends BWExpression {

    private final String className;
    private BWObject object = new BWInteger(0);

    public ObjectCreation(int lineNumber, String className) {
        super(lineNumber);
        this.className = className;
    }

    @Override
    public BWObject getValue() {
        return object;
    }

    @Override
    public int getIntValue() {
        return object.getIntValue();
    }

    @Override
    public String getValueType() {
        return object.getValueType();
    }

    @Override
    public void run(Scope scope) {
        BWClass bwClass = BWClass.getClassForName(className);
        object = bwClass.createInstance();
    }
}

File: src/main/java/org/dedda/bratwurst/lang/ObjectCreation.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.classes.BWClass
import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
class ObjectCreation(lineNumber: Int, private val className: String) : BWExpression(lineNumber) {
    private var obj: BWObject = BWInteger(0)
    override fun getValue(): BWObject {
        return obj
    }

    override fun getIntValue(): Int {
        return obj.intValue
    }

    override fun getValueType() = obj.valueType

    override fun run(scope: Scope) {
        val bwClass = BWClass.getClassForName(className)
        obj = bwClass.createInstance()
    }

}